lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -93,6 +93,27 @@ OpFoldResult ConstantOp::fold(ArrayRef<Attribute> operands) {
 // Binary Operators
 //===----------------------------------------------------------------------===//
 
+OpFoldResult AddPrimOp::fold(ArrayRef<Attribute> operands) {
+  /// Any folding here requires a bitwidth extension.
+
+  /// If both operands are constant, and < 64 bits, then perform constant
+  /// folding. Cannot use constFoldBinaryOp, since the width of the constant is
+  /// different from operands.
+  if (operands[0] && operands[1] &&
+      lhs().getType().cast<FIRRTLType>().getBitWidthOrSentinel() < 64) {
```

Comment:
You shouldn't need to limit this to 64 bits (see below).

However, we have to be super careful about add operations that have operands/results that have unknown widths.  I'd recommend defining a general predicate that can be used by other folders so you can do something like this:

`  if (operands[0] && operands[1] && hasKnownWidths(this))`

hasKnownWidths would check that the operands and results of the Operation* are all integer types with known widths.